### PR TITLE
[Modules] Updated ContainerInstances ContainerGroups to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated ContainerInstances ContainerGroups to new dependency approach

| Pipeline |
| - |
| [ContainerInstance: ContainerGroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerinstance.containergroups.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.ContainerInstance/containerGroups/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,62 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.ContainerInstance/containerGroups/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

**Review by eriqua — DISMISSED** [review]
🆗

File: modules/Microsoft.ContainerInstance/containerGroups/.test/min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    lock: 'CanNotDelete'
```
Comment (eriqua) [inline modules/Microsoft.ContainerInstance/containerGroups/.test/min/deploy.test.bicep:36]: I'd suggest to only include required parameters in the min test. So to remove `lock` and `ipAddressPorts`.

Comment (MrMCake) [reply modules/Microsoft.ContainerInstance/containerGroups/.test/min/deploy.test.bicep]: In the given configuration, the IPAddressPorts are actually required. However, I don't know the service well enough to judge if that makes it a generally required parameter. Lock is removed

Comment (eriqua) [reply modules/Microsoft.ContainerInstance/containerGroups/.test/min/deploy.test.bicep]: I'll then open an issue to investigate on it and approve meanwhile

Comment (eriqua) [reply modules/Microsoft.ContainerInstance/containerGroups/.test/min/deploy.test.bicep]: Created issue #2205.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**